iceoryx_binding_c/source/c_publisher.cpp

```diff
@@ -34,21 +34,20 @@ iox_pub_t iox_pub_init(iox_pub_storage_t* self,
                        const char* const service,
                        const char* const instance,
                        const char* const event,
-                       const uint64_t historyCapacity,
-                       const char* const nodeName)
+                       const iox_publisher_options_t options)
 {
     new (self) cpp2c_Publisher();
     iox_pub_t me = reinterpret_cast<iox_pub_t>(self);
-    PublisherOptions options;
-    options.historyCapacity = historyCapacity;
-    options.nodeName = NodeName_t(TruncateToCapacity, nodeName);
+    PublisherOptions publisherOptions;
+    publisherOptions.historyCapacity = options.historyCapacity;
+    publisherOptions.nodeName = NodeName_t(TruncateToCapacity, options.nodeName);
```

Comment:
Maybe we should print a warning here when `options.nodeName` is longer then the `NodeName_t` capacity so that the user is at least informed and does not report a bug that his node name is truncated. Or we follow the array approach.

Reply:
isn't `cxx::string` already printing a warning when the string is truncated?

Reply:
I think so but the warning is somehow out of context since the user does not know that we are using a `cxx::string` behind the scenes. It is better then nothing but from a user point of view I would wish to get as much information as possible so that I can fix the problem.

An error message like: `publisher node name to long, truncate it to: "truncatedNodeName"` is very informative and would me help a lot!

Reply:
The cxx string prints the input parameter when giving a warning that it will be truncated.
```bash
Constructor truncates the last 1 characters of TooLongNodeName, because the char array length is larger than the capacity.
```

Reply:
Yes but we could argue this for a lot of parameters and would need a systematic approach on validation (is the historycapacity valid? Too large? Do we truncate or fail? How do we inform the user). I would not do a check here in addition to the cxx::string since it I consider the additional value low.